yarn-project/simulator/src/avm/opcodes/instruction.ts

```diff
@@ -78,7 +98,16 @@ export abstract class Instruction {
       deserialize: (buf: BufferCursor | Buffer): Instruction => {
         const res = deserialize(buf, wireFormat);
         const args = res.slice(1); // Remove opcode.
-        return new this(...args);
+        const instance = new this(...args);
+        // Need to make sure that deserialized instances can getSize
+        return Object.defineProperties(instance, {
+          getSize: {
+            value: (): number => {
+              return getInstructionSize(wireFormat);
+            },
+            enumerable: false,
+          },
+        });
```

Comment:
This is ugly.... Is there some way I can get around needing to do this `Object.defineProperties()` INSIDE an `Object.assign()`. I'm having trouble thinking of better ways to handle this.